### PR TITLE
Remove support for 1.x product-configure schema

### PR DESCRIPTION
**Pull request by bmerry**
2.x has been around for a long time now and is described in the ICD, so
there is little reason to hang on to 1.x, and getting rid of it will
reduce the complexity of a 3.x I want to introduce.

### CODE REVIEW

**Comment by ajoubertza** [comment]
Fine with me.  CAM is using schema version 2.4 (or at least claims to!)

https://github.com/ska-sa/katproxy/blob/0b3d20db7bddc5b23ab8a8f73c7e1e31e9743f2b/katproxy/proxy/data_sdp_manager.py#L673

File: katsdpcontroller/schemas/product_config.json.j2 [hunk 1]
```diff
@@ -6,19 +6,13 @@
     "properties": {
         "version": {
             "type": "string",
-            "enum": ["1.0", "1.1", "2.0", "2.1", "2.2", "2.3", "2.4", "2.5", "2.6"]
```
Comment (sratcliffe) [inline katsdpcontroller/schemas/product_config.json.j2:9]: Any impact to retiring 1.x support ? (I suspect not, but need to ask)

Comment (sratcliffe) [reply katsdpcontroller/schemas/product_config.json.j2:9]: Happy with that - if we are going to cull we probably should cull to the latest version...

Comment (bmerry) [reply katsdpcontroller/schemas/product_config.json.j2:9]: I'll remove pre-2.0 for now. Trying to remove 2.0-2.3 in this PR will probably conflict with another branch I'm working on, but I'll take this as license to drop support for them in a later PR.

**Pull request merged**